aten/src/ATen/native/Sorting.cpp

```diff
@@ -94,6 +91,169 @@ void quick_select_template(
   } while (true);
 }
 
+// Returns pointer to pivot used to partition range [first, last]
+// There should be at least 3 elements in [first, last]
+template <typename random_it, typename comp>
```

Comment:
Do you want to add a brief description of the algorithm you're using, too?

Reply:
Also: what are the invariants the arguments to this function need to satisfy, and what are the invariants coming out of it?

Reply:
Second question: how does this algorithm compare to std::partition?